src/controller/src/lib.rs

```diff
@@ -217,7 +217,8 @@ where
     /// `source_ids` at the time of the function call.
     #[allow(unused)]
     #[allow(clippy::unused_async)]
-    pub async fn recent_timestamp(&self, source_ids: &[GlobalId]) -> T {
+    pub async fn recent_timestamp(&self, source_ids: impl Iterator<Item = GlobalId>) -> T {
```

Comment:
@petrosagg Are you ok with this API change?

Reply:
We don't yet have an implementation so.. tentative yes. Once we start wiring up the storage side we can see if we need to bring back the slice. Worst case we collect into a vec. Feel free to merge it for now